### ISSUE TITLE
Error with slog macros

### ISSUE THREAD

**Issue by thomas-jeepe** [intellij-rust/intellij-rust#1600]
Using the slog `error!` macro causes an error within the IDE for an invalid syntax error. The example is below.

`Cargo.toml`

```toml
[package]
name = "thing"
version = "0.1.0"
authors = [""]

[dependencies]
cdrs = "1.0.0-rc.3"
slog = "2.0.6"
slog-json = "2.0.1"
```

`src/main.rs`

```rust
#[macro_use]
extern crate slog;
extern crate slog_json;

use slog::Drain;
use std::sync::Mutex;

use slog::{Result, KV, Key, Record, Serializer};

struct XD {
    haha: u8,
    oh_no: u64
}

impl KV for XD {
    fn serialize(&self, record: &Record, serializer: &mut Serializer) -> Result {
        serializer.emit_u8("haha", self.haha)?;
        serializer.emit_u64("oh_no", self.oh_no)?;
        Ok(())
    }
}

fn main() {
    let root = slog::Logger::root(
        Mutex::new(slog_json::Json::default(std::io::stderr())).map(slog::Fuse),
        o!("version" => env!("CARGO_PKG_VERSION"))
    );

    error!(root, "yikes"; XD { haha: 1, oh_no: 2 });
}
```

Specifically the error I am getting is on the `error!` macro line, I got:

`'(', '[' or '{' expected, got '('`

Extra:

Using the `slog_error!` macro instead of the `error!` causes the ide to not error.

Using either `slog_error!` or `error!` both compile and work with `cargo run`.

```
IntelliJ IDEA 2017.2.1
Build #IC-172.3544.35, built on July 31, 2017
JRE: 1.8.0_152-release-915-b6 x86_64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Mac OS X 10.11.6
```

I don't know how to get the plugin version.

**Comment by Deedasmi** [comment]
Still happens on 0.2.0.2076.

Additionally, errors inside the macro (on variable handling) with 
![image](https://user-images.githubusercontent.com/5093293/32530416-b3adfaac-c403-11e7-9dc9-55e6bc143e39.png)

For some example code it happens on a line like:
```
info!(peer_log, "New WebSocket "; "connection " => addr.to_string());
```
in 3 places, the semi-colon between string literal and connection, the space before the equals sign, and the final closing parenthesis.

**Comment by Undin** [comment]
Fixed in #2608
